frontend/src/stores/useBiolink.js

```diff
@@ -0,0 +1,123 @@
+import { useState, useEffect } from 'react';
+import _ from 'lodash';
+import strings from '~/utils/strings';
+
+const baseClass = 'biolink:BiologicalEntity';
+
+export default function useBiolink() {
+  const [biolink, setBiolink] = useState(null);
+  const [concepts, setConcepts] = useState([]);
+  const [hierarchies, setHierarchies] = useState({});
+  const [predicates, setPredicates] = useState([]);
+
+  function getEdgePredicates() {
+    const newPredicates = Object.entries(biolink.slots).map(([identifier, predicate]) => ({
+      predicate: strings.edgeFromBiolink(identifier),
+      domain: strings.nodeFromBiolink(predicate.domain),
+      range: strings.nodeFromBiolink(predicate.range),
+    }));
+    return newPredicates;
+  }
+
+  /**
+   * Given a biolink class, build a list of parent classes
+   * Ex: Given 'gene' as an input this function will return:
+   *
+   * [ "gene", "gene_or_gene_product", "macromolecular_machine",
+   *   "genomic_entity", "molecular_entity", "biological_entity",
+   *   "named_thing" ]
+   *
+   * @param {string} className - a biolink class
+   * @param {object} biolinkClasses - object of all biolink classes
+   * @returns {string[]} list of related biolink classes
+   */
+  function getAncestors(childClass, classes) {
+    let currentClass = childClass;
+    const ancestors = [currentClass];
+    // Repeat until we hit the top of the classes
+    while (
+      classes[currentClass] &&
+      classes[currentClass].is_a
+    ) {
+      // reassign current type to parent type
+      currentClass = classes[currentClass].is_a;
+      ancestors.push(currentClass);
+    }
+    return ancestors;
+  }
+
+  /**
+   * Get a list of descendants for all biolink classes
```

Comment:
This function is named getAncestries but the description says "list of descendants". Are those the same thing?

Reply:
Fixed in 3b264f7